calico-vpp-agent/config/config.go

```diff
@@ -139,12 +139,19 @@ func getEnvValue(str string) string {
 }
 
 func fetchHostMtu() (mtu int, err error) {
+	vppManagerInfo := &vppmanagerconfig.VppManagerInfo{}
 	for i := 0; i < 20; i++ {
-		dat, err := ioutil.ReadFile(VppManagerLinuxMtu)
+		dat, err := ioutil.ReadFile(VppManagerInfoFile)
 		if err == nil {
-			idx, err := strconv.ParseInt(strings.TrimSpace(string(dat[:])), 10, 32)
-			if err == nil && idx != -1 {
-				return int(idx), nil
+			err2 := json.Unmarshal(dat, vppManagerInfo)
+			if err2 == nil && len(vppManagerInfo.Uplinks) != 0 {
+				minMtu := 1000000
+				for _, v := range vppManagerInfo.Uplinks {
+					if v < minMtu {
```

Comment:
But this logic should probably go in the `GetUplinkMtu()` function (json will default to 0, so we probably want to log an error and use `vpplink.MAX_MTU` if this happens)